lib/block_view_test.go

```diff
@@ -61,6 +67,665 @@ var (
+// to the mempool. Lastly, _testVectorDependency keeps track of all the test vectors which should be skipped from checking
+// verifyConnectUtxoViewEntry and verifyDbEntry because there has been a later test vector (test vector identified by map key id) that
+// had a pairwise IsDependency evaluate to true. This is used to avoid checking the same utxoView or db records repeatedly:
+// once for the latest testVector, and twice, thrice, etc. for the earlier, dependent testVectors.
+type transactionTestSuite struct {
+	t                *testing.T
+	testVectorBlocks []*transactionTestVectorBlock
+	config           *transactionTestConfig
+
+	_testVectorsInDb      []*transactionTestVector
+	_testVectorsInMempool []*transactionTestVector
+	_testVectorDependency map[transactionTestIdentifier][]transactionTestIdentifier
+}
+
+func NewTransactionTestSuite(t *testing.T, testVectorBlocks []*transactionTestVectorBlock,
+	config *transactionTestConfig) *transactionTestSuite {
+
+	return &transactionTestSuite{
+		t:                     t,
+		testVectorBlocks:      testVectorBlocks,
+		config:                config,
+		_testVectorDependency: make(map[transactionTestIdentifier][]transactionTestIdentifier),
+	}
+}
+
+// Run is the main method of the transactionTestSuite. It will run the test suite and verify that all the testVectors are
+// pass a variety of tests.
+func (tes *transactionTestSuite) Run() {
+	// Make sure all test vectors have unique Ids.
+	tes.ValidateTestVectors()
+
+	if tes.config.testBadger {
+		tes.RunBadgerTest()
+	}
+	if tes.config.testPostgres {
+		tes.RunPostgresTest()
+	}
+}
+
+// Iterate over all testVectors and ensure all of them have unique Ids.
+func (tes *transactionTestSuite) ValidateTestVectors() {
+	testVectorsIds := make(map[transactionTestIdentifier]struct{})
+	for _, testVectorBlocks := range tes.testVectorBlocks {
+		for _, testVectors := range testVectorBlocks.testVectors {
+			if _, exists := testVectorsIds[testVectors.id]; exists {
+				tes.t.Fatalf("Duplicate test vector id: %v", testVectors.id)
+			}
+			testVectorsIds[testVectors.id] = struct{}{}
+		}
+	}
+}
+
+func (tes *transactionTestSuite) RunBadgerTest() {
+	glog.Infof(CLog(Yellow, "RunBadgerTest: TESTING BADGER"))
+
+	tm := tes.InitializeChainAndGetTestMeta(true, false)
+	tes._testVectorsInDb = []*transactionTestVector{}
+	tes._testVectorsInMempool = []*transactionTestVector{}
+	tes._testVectorDependency = make(map[transactionTestIdentifier][]transactionTestIdentifier)
+
+	for _, testVectorBlock := range tes.testVectorBlocks {
+		// Run all the test vectors for this block.
+		dbEntriesBefore := tes.GetAllStateDbEntries(tm)
+		tes.testConnectBlock(tm, testVectorBlock)
+		tes.testDisconnectBlock(tm, testVectorBlock)
+		dbEntriesAfter := tes.GetAllStateDbEntries(tm)
+		tes.compareBeforeAfterDbEntries(dbEntriesBefore, dbEntriesAfter)
+		glog.Infof(CLog(Yellow, "RunBadgerTest: successfully connected/disconnected block and verified db state"))
```

Comment:
Can we set a glog level such that our builds don't have all these print statements in them please? applies to all print statements.